crates/sequencer/src/sequencer.rs

```diff
@@ -954,6 +969,68 @@ where
         }
     }
 
+    async fn process_l1_block(
+        &self,
+        l1_block: <Da as DaService>::FilteredBlock,
+    ) -> anyhow::Result<()> {
+        let mut sequencer_commitments = Vec::<SequencerCommitment>::new();
+
+        self.da_service
+            .extract_relevant_blobs(&l1_block)
+            .into_iter()
+            .for_each(|mut tx| {
+                let data = DaData::try_from_slice(tx.full_data());
+                // Check for commitment
+                if tx.sender().as_ref() == self.sequencer_pub_key.as_slice() {
+                    if let Ok(DaData::SequencerCommitment(seq_com)) = data {
+                        sequencer_commitments.push(seq_com);
+                    } else {
+                        tracing::warn!(
+                            "Found broken DA data in block 0x{}: {:?}",
+                            hex::encode(l1_block.hash()),
+                            data
+                        );
+                    }
+                }
+            });
```

Comment:
I don't think we should scan L1 like  this in the sequencer.

we want to remove the submission from the pending list as soon as the tx lands in the L1 mempool so that if the sequencer restarts again it wont resubmit the commitment. 


again, after #537 this will be handled much better

Reply:
We are receiving L1 events anyway... we might as well use this to:
1. Confirm a commitment has been included in a new DA block
2. Remove the L2 ranges of those commitments from pending
3. Update PG to set the status of a commitment as mined.

The current way of removing a range from pending is blind (There's no confirmation the tx has been mined, could potentially stay in the mempool for a long time if not dropped).

Reply:
> There's no confirmation the tx has been mined, could potentially stay in the mempool for a long time if not dropped

we will handle that with RBF or CPFP on the bitcoin da adapter level.

> so that if the sequencer restarts again it wont resubmit the commitment.

the problem is we are not gonna be able to replace a commitment that has been sent, we will be double sending the commitment